Doxygen/man/man3/GPIB_pyvisa.3
.TH "GPIB_pyvisa" 3 "Tue Oct 5 2021" "AutoWave control python library" \" -*- nroff -*-
.ad l
.nh
.SH NAME
GPIB_pyvisa
.SH SYNOPSIS
.br
.PP
.SS "Functions"

.in +1c
.ti -1c
.RI "def \fBdelay_cmd\fP (time_s=2)"
.br
.ti -1c
.RI "def \fBpsend\fP (txt_cmd)"
.br
.in -1c
.SS "Variables"

.in +1c
.ti -1c
.RI "list \fBAG_status\fP"
.br
.ti -1c
.RI "\fBrm\fP = pyvisa\&.ResourceManager()"
.br
.ti -1c
.RI "\fBrm_list\fP = rm\&.list_resources()"
.br
.ti -1c
.RI "int \fBi\fP = 0"
.br
.ti -1c
.RI "string \fBresource_name\fP = ''"
.br
.ti -1c
.RI "\fBinst\fP = rm\&.open_resource(\fBresource_name\fP)"
.br
.ti -1c
.RI "\fBwrite_termination\fP"
.br
.ti -1c
.RI "\fBfile_list\fP = inst\&.write_raw(\fBpsend\fP('DIR? /home/guest/DowFiles'))"
.br
.ti -1c
.RI "\fBfile_list_array\fP = file_list\&.split(',')"
.br
.ti -1c
.RI "int \fBnumber\fP = 0"
.br
.ti -1c
.RI "string \fBtest_file_name\fP = 'Ford ES\-XW7T\-1A278\-AC \- CI210 \-\&.dsg'"
.br
.ti -1c
.RI "\fBstatus\fP = inst\&.query('STAT? TEST')"
.br
.RI "requires to send double start\&. "
.ti -1c
.RI "\fBst\fP = status\&.replace('STAT TEST:', '')"
.br
.in -1c
.SH "Function Documentation"
.PP 
.SS "def GPIB_pyvisa\&.delay_cmd ( time_s = \fC2\fP)"

.PP
Definition at line \fB24\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "def GPIB_pyvisa\&.psend ( txt_cmd)"

.PP
Definition at line \fB27\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SH "Variable Documentation"
.PP 
.SS "list GPIB_pyvisa\&.AG_status"
\fBInitial value:\fP
.PP
.nf
1 =  ["stopped",
2             "ready",
3             "started",
4             "fail",
5             "fast",
6             "DSP not ok",
7             "break",
8             "not ready",
9             "finished",
10             "iterate",
11             "undefined",
12             "write to file",
13             "file processing",]
.fi
.PP
Definition at line \fB9\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "GPIB_pyvisa\&.file_list = inst\&.write_raw(\fBpsend\fP('DIR? /home/guest/DowFiles'))"

.PP
Definition at line \fB96\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "GPIB_pyvisa\&.file_list_array = file_list\&.split(',')"

.PP
Definition at line \fB98\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "int GPIB_pyvisa\&.i = 0"

.PP
Definition at line \fB59\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "GPIB_pyvisa\&.inst = rm\&.open_resource(\fBresource_name\fP)"

.PP
Definition at line \fB68\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "int GPIB_pyvisa\&.number = 0"

.PP
Definition at line \fB99\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "GPIB_pyvisa\&.resource_name = ''"

.PP
Definition at line \fB60\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "GPIB_pyvisa\&.rm = pyvisa\&.ResourceManager()"

.PP
Definition at line \fB57\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "GPIB_pyvisa\&.rm_list = rm\&.list_resources()"

.PP
Definition at line \fB58\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "GPIB_pyvisa\&.st = status\&.replace('STAT TEST:', '')"

.PP
Definition at line \fB127\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "GPIB_pyvisa\&.status = inst\&.query('STAT? TEST')"

.PP
requires to send double start\&. \&. to trigger genetation 
.PP
Definition at line \fB126\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "string GPIB_pyvisa\&.test_file_name = 'Ford ES\-XW7T\-1A278\-AC \- CI210 \-\&.dsg'"

.PP
Definition at line \fB104\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SS "GPIB_pyvisa\&.write_termination"

.PP
Definition at line \fB74\fP of file \fBGPIB_pyvisa\&.py\fP\&.
.SH "Author"
.PP 
Generated automatically by Doxygen for AutoWave control python library from the source code\&.


src/GPIB_pyvisa.py
##
##
##

import sys
import pyvisa
import time

AG_status = ["stopped",
            "ready",
            "started",
            "fail",
            "fast",
            "DSP not ok",
            "break",
            "not ready",
            "finished",
            "iterate",
            "undefined",
            "write to file",
            "file processing",]


def delay_cmd(time_s=2):
    time.sleep(time_s)

def psend(txt_cmd):
    print("Psend input:", txt_cmd)
    sum = 0 # variable for check summ
    cmd = [] # array for formatted command

    for item in txt_cmd:
        # sum = sum + hex(item.encode('utf-8').hex())
        #print(f"Symbol: {item}, value: {ord(item)}")
        sum = sum + int(ord(item))
        cmd.append(ord(item))
    check_sum = sum & 0x00FF
    print(f"sum: {sum}, check_sum: {check_sum}, cmd: {cmd}")
    #  If the checksum is less or equal to 0x20, 0x20 is added again.
    #  Thus ensures that the checksum is not interpreted as control character.
    if check_sum <= 0x20:
        check_sum += 0x20
    # additional protocol requirements
    # STX=0x02 + Command + ETX=0x03 + CheckSum
    cmd.insert(0, 2)  # insertion of STX=0x02 to a first place
    cmd.append(3)  # termination message with ETX=0x03
    cmd.append(check_sum)  # adding check sum at the end
    print(f"protocol cmd: {cmd}")
    print(bytearray(cmd))
    return (bytes(cmd))
    # return ((cmd))





rm = pyvisa.ResourceManager()
rm_list = rm.list_resources()
i = 0
resource_name = ""
for item in rm_list:
     if "AutoWave" in item:
         resource_name = item
     # print(f"{i} : {item}")
     i = i+1
# resource_name = "USB0::0x03EB::0x2065::_IDN_EM_TEST__AutoWave__0__5.10.08__2__0::INSTR"
if resource_name != "":
    inst = rm.open_resource(resource_name)
    print(inst)
inst.set_visa_attribute(pyvisa.constants.VI_ATTR_SEND_END_EN, 1)
print(f'VI_ATTR_SEND_END_EN={inst.get_visa_attribute(pyvisa.constants.VI_ATTR_SEND_END_EN)}')
#print(f'VI_ATTR_ASRL_END_OUT={inst.get_visa_attribute(pyvisa.constants.VI_ATTR_ASRL_END_OUT)}')
# print("write_termination:", inst.write_termination)
inst.write_termination = ""
print("write_termination:", inst.write_termination)
# inst.read_termination = '\n'
# inst.write_termination = '\n'
#inst.baud_rate = 9600

inst.write("*PRCL:ON")
print("Session:", inst.session)
delay_cmd()
print(inst.query("*IDN?"))
delay_cmd()
print(inst.query("*PRCL?"))
delay_cmd()
print(inst.query("*ECHO:ON"))
delay_cmd()
# print(inst.query_binary_values(b'\x02STAT? SYST\x03\xee'))
print(inst.write_raw(psend("STAT? SYST")))
print(inst.read_raw())
delay_cmd()
# inst.write("DISP Python control")
# delay_cmd()
print("*** TESTING INIT ***")
file_list = inst.write_raw(psend("DIR? /home/guest/DowFiles"))
print(file_list)
file_list_array = file_list.split(",")
number = 0
for item in file_list_array:
    print(f'{number} : {item}')
    number = number +1
delay_cmd()
test_file_name = "Ford ES-XW7T-1A278-AC - CI210 -.dsg"
print("TEST file:", inst.query(f"CKLF? {test_file_name}"))
delay_cmd()
print(inst.query(f"TRFL home/guest/DowFiles/{test_file_name}"))
delay_cmd()
print(inst.query("MOD GEN"))
delay_cmd()
print(inst.query(f"SOUR SEGM {test_file_name}"))
delay_cmd()
delay_cmd()
print(inst.query("TRIG:GEN 1")) # 1 is for a manual stat on cmd
delay_cmd()
##requires to send double start.. to trigger genetation
print(inst.query("STAR")) #trigger the test start
delay_cmd()
print(inst.query("STAR")) #trigger the test start


delay_cmd()
for i in range(10):
    delay_cmd()
    delay_cmd()
    status = inst.query("STAT? TEST")
    st = status.replace("STAT TEST:", "")
    st = st.split(",")
    print("st:", st)
    for code in st:

        print(AG_status[int(code)])

# IDN = str(inst.query("*IDN?"))
# print(f': Connected to: {IDN}')

# print(app.query("OUT?"))
#
# app.write("OUT ON")
# time.sleep(0.25)
# print(app.query("OUT?"))
# time.sleep(0.25)
# app.write("ISET 10A")
# time.sleep(2)
# app.write("VSET 13.5V")
# print(app.query("VSET?"))
# time.sleep(2)
# app.write("OUT OFF")
# print(app.query("OUT?"))